src/common/io.cc

```diff
@@ -120,27 +121,22 @@ std::string LoadSequentialFile(std::string fname) {
 #if defined(__linux__)
   posix_fadvise(fd, 0, 0, POSIX_FADV_SEQUENTIAL);
 #endif  // defined(__linux__)
-  ssize_t bytes_read = read(fd, &buffer[0], f_size_bytes);
```

Comment:
@trivialfis Previously, we called `read()` only once. For a file containing 2896075944 bytes (2.69 GB), `read()` only reads 2147479552 bytes (1.99 GB), and the variables would be set as follows:
```
bytes_read = 2147479552, f_size_bytes = 2896075944
```

This PR ensures that `bytes_read` would be identical to `f_size_bytes`.